lib/private_api.rb

```diff
@@ -76,86 +117,56 @@ def format_order_hash(o)
 
 
 
-
-  # <----- BALANCE .
-
-  #Balance
-  def balance_for(asset='XBT', opt={})
-    puts "BitX.balance_for will be deprecated. please use BitX.balance to get a list of account balances"
-    opt.merge!({asset: asset})
-    path = '/api/1/balance'
-    r = authed_request(path, {params: opt})
-    j = JSON.parse(r.body)
-    raise BitXError.new("BitX balance error: #{j['error']}") if j['error']
-    balance = BigDecimal(j['balance'][0]['balance'])
-    reserved = BigDecimal(j['balance'][0]['reserved'])
-    {
-      asset:      asset,
-      balance:    balance,
-      reserved:   reserved,
-      available:  balance - reserved
+  # <----- TRANSACTIONS .
+  def transactions(account_id, min_row=1, max_row=100)
+    path = "https://api.mybitx.com/api/1/accounts/#{account_id}/transactions"
+    opt = {
+      min_row: min_row,
+      max_row: max_row
     }
+    authed_request(path, opt)
   end
 
-  def balance(opt={})
-    path = '/api/1/balance'
-    r = authed_request(path, opt)
-    j = JSON.parse(r.body)
-
-    balances = []
-    if j['balance']
-      j['balance'].each do |b|
-        balance     = BigDecimal(b['balance'])
-        reserved    = BigDecimal(b['reserved'])
-        balances << {
-          account_id:  b['account_id'],
-          asset:       b['asset'],
-          balance:     balance,
-          reserved:    reserved,
-          available:   balance - reserved,
-          unconfirmed: BigDecimal(b['unconfirmed'])
-        }
-      end
-    end
-    balances
+  def pending(account_id)
+    authed_request("https://api.mybitx.com/api/1/accounts/#{account_id}/pending")
   end
+  # TRANSACTIONS -----/>
 
-  # BALANCE -----/>
 
 
 
-  # <----- receive addresses .
+  # <----- RECEIVE ADDRESSES .
   def new_receive_address(opt={})
     receive_address_request(opt, :post)
   end
 
   # if you have multiple XBT accounts you can specify an `address` option with the funding address
-  def received_by_address(opt={})
-    receive_address_request(opt, :get)
-  end
-
-  def receive_address_request(opt, method)
-    opt.merge!({asset: 'XBT'})
-    path = '/api/1/funding_address'
-    r = authed_request(path, {params: opt, method: method})
-    JSON.parse(r.body, {symbolize_names: true})
+  def received_by_address(address=nil)
+    if address.is_a? Hash
+      address = address[:address]
+    end
+    receive_address_request({address: address}, :get)
   end
 
   #Bitcoin Funding Address
   def funding_address(asset='XBT', opt={})
-    puts "BitX.funding_address will be deprecated. please use BitX.received_by_address with a specified :address option to get details for that address"
+    puts "BitX.funding_address will be deprecated. please use BitX.received_by_address with a specified address paramenter or an :address option to get details for that address"
     opt.merge!({asset: asset})
     receive_address_request(opt, :get)
   end
-  # receive addresses -----/>
+
+  def receive_address_request(opt, method)
```

Comment:
It seems dangerous to allow the caller to specify an HTTP method... Maybe this should be split into get_receive_address and create_receive_address?